docs/CONTRIBUTING.md

```diff
@@ -78,26 +81,44 @@ we'll guide you.
 
 ## Development workflow
 
-The current development workflow is a bit tricky because we have this repo and we use our `coder/vscode` fork inside it with [`yarn link`](https://classic.yarnpkg.com/lang/en/docs/cli/link/).
-
-Here are these steps you should follow to get your dev environment setup:
-
 1. `git clone https://github.com/coder/code-server.git` - Clone `code-server`
 2. `git submodule update --init` - Clone `vscode` submodule
 3. `yarn` - Install dependencies
-4. `yarn watch` - This will spin up code-server on localhost:8080 which you can start developing. It will live reload changes to the source.
+4. `quilt patch -a` - Apply patches to the `vscode` submodule.
+5. `yarn watch` - Launch code-server localhost:8080. code-server will be live
+   reloaded when changes are made; the browser needs to be refreshed manually.
 
-### Updates to VS Code
+When pulling down changes that include modifications to the patches you will
+need to apply them with `quilt`. If you pull down changes that update the
+`vscode` submodule you will need to run `git submodule update --init` and
+re-apply the patches.
 
-If changes are made and merged into `main` in the [`coder/vscode`](https://github.com/coder/vscode) repo, then you'll need to update the version in the `code-server` repo by following these steps:
+### Version updates to Code
 
-1. Update the `lib/vscode` submodule to the latest `main`.
+1. Update the `lib/vscode` submodule to the desired upstream version branch.
 2. From the code-server **project root**, run `yarn install`.
-3. Test code-server locally to make sure everything works.
-4. Check the Node.js version that's used by Electron (which is shipped with VS
+3. Apply the patches (`quilt push -a`) or restore your stashed changes. At this
+   stage you may need to resolve conflicts. For example use `quilt push -f`,
+   manually apply the rejected portions, then `quilt refresh`.
+4. Test code-server locally to make sure everything works.
+5. Check the Node.js version that's used by Electron (which is shipped with VS
    Code. If necessary, update your version of Node.js to match.
-5. Commit the updated submodule to `code-server`.
-6. Open a PR.
+6. Commit the updated submodule and patches to `code-server`.
+7. Open a PR.
+
+### Patching Code
+
+0. You can go through the patch stack with `quilt push` and `quilt pop`.
+1. Create a new patch (`quilt new {name}.diff`) or use an existing patch.
+2. Add the file(s) you are patching (`quilt add [-P patch] {file}`). A file
+   **must** be added before you make changes to it.
+3. Make your changes. Patches do not need to be independent of each other but
+   each patch must result in a working code-server without any broken in-between
```

Comment:
Very glad you added the "each patch must result in a working code-server without any broken in-between" 👏🏼

Reply:
Gotta have a working `git bisect`!!